lib/cassanity/column_family.rb

```diff
@@ -213,6 +213,23 @@ def select(args = {})
       })
     end
 
+    # Public: Makes it possible to asynchronously query data from the column family.
+    #
+    # args - The Hash of arguments to pass to the argument generator
+    #        (default: {}). :column_family_name and :keyspace_name are
+    #        always included.
+    #
+    # Returns whatever is returned by executor.
+    def select_async(args = {})
+      @executor.call({
```

Comment:
I wonder if we should have executor and async_executor. That is my only thought. Did you consider this? Making a single executor responsible for both seems like a lot in an already kind of overloaded area.